src/test/regress/sql/columnar_create.sql

```diff
@@ -65,9 +65,15 @@ FROM pg_class WHERE relname='columnar_temp' \gset
 
 \c - - - :master_port
 
--- show that temporary table itself and it's metadata is removed
-SELECT COUNT(*)=0 FROM pg_class WHERE relname='columnar_temp';
-SELECT columnar_test_helpers.columnar_metadata_has_storage_id(:columnar_temp_storage_id);
+-- Show that temporary table itself and it's metadata is removed, or
+-- table isn't dropped yet and metadata records do also persist.
+--
+-- This might happen since \c doesn't guarantee immediate cleanup of
+-- all the resources.
+SELECT columnar_test_helpers.columnar_metadata_has_storage_id(:columnar_temp_storage_id)=q.pg_class_has_table
+FROM (
+  SELECT count(*)!=0 AS pg_class_has_table FROM pg_class WHERE relname='columnar_temp'
+) q;
```

Comment:
There's still technically a race here. Also, this weakens the test because the idea was to be sure the temp tables are removed.

I'd suggest either we find some way to wait on the other backend pid, which would probably be a hack, or we just sleep and document why the sleep is there. Probably a second or three is enough to make the test not flakey.

The "right" way would be to introduce a new function `pg_waitpid()`, and it would wait for another backend to expire. But it seems a little too much here.

Reply:
pushed some changes for that